node/src/blockchain/blockchain_interface.rs

```diff
@@ -384,10 +412,58 @@ where
 
 impl<T> BlockchainInterfaceNonClandestine<T>
 where
-    T: Transport + Debug + 'static,
+    T: 'static + BatchTransport + Debug,
+{
+    fn find_largest_transaction_block_number(
+        &self,
+        response_block_number: u64,
+        transactions: &Vec<BlockchainTransaction>,
+    ) -> u64 {
+        if transactions.is_empty() {
+            response_block_number
+        } else {
+            transactions
+                .iter()
+                .fold(response_block_number, |so_far, elem| {
+                    if elem.block_number > so_far {
+                        elem.block_number
+                    } else {
+                        so_far
+                    }
+                })
+        }
+    }
+}
+
+impl<T> BlockchainInterfaceNonClandestine<T>
+where
+    T: 'static + BatchTransport + Debug,
+{
+    fn extract_transactions_from_logs(&self, logs: Vec<Log>) -> Vec<BlockchainTransaction> {
+        logs.iter()
+            .filter_map(|log: &Log| match log.block_number {
+                None => None,
+                Some(block_number) => {
+                    to_gwei(U256::from(log.data.0.as_slice())).map(|gwei_amount| {
```

Comment:
Maybe this idea is silly but what if you put the value `log.data.0` on a different line and give it a name through using a variable. Readability of the code might get better. I'll let you decide if it's worth it. But you know, only the person who will go to the source code of the library will really know what this is exactly.